Async handler improvements

danielmarbach opened the pull request:
Closes #2693

cremor commented on nservicebus/handlers/async-handlers.md:
```diff
@@ -11,21 +11,23 @@ versions: '[6.0,)'
 WARNING: It is difficult to give generic advice on how asynchronous code should be structured. It is important to understand compute-bound vs. IO-bound operations. Avoid copy-pasting snippets without further analysis if they provide benefit for the involved business scenarios. Don't assume. Measure it.
 
-[Handlers](/nservicebus/handlers/) and [Sagas](/nservicebus/sagas/) will be invoked from an IO thread pool thread. Typically message handlers and sagas issue IO bound work like sending or publishing messages, storing information into databases, calling web services and more. In other cases, message handlers are used to schedule compute-bound work. To be able to write efficient message handlers an ...
+[Handlers](/nservicebus/handlers/) and [Sagas](/nservicebus/sagas/) will be invoked from from a thread pool thread. Depending on the transport implementation it might use a worker thread pool thread or an IO thread pool thread. Typically message handlers and sagas issue IO bound work like sending or publishing messages, storing information into databases, calling web services and more. In other c ...
```
Duplicate "from" in first sentence.

I'm not sure if the "it" in the second sentence makes sense. (Yes I know that I wrote that sentence originally :smile:) Also, now that the two thread pools were rewritten, maybe this should be updated too. How about "Depending on the transport implementation this thread might be associated with compute bound work or IO-bound work."

danielmarbach commented:
@cremor addressed your feedback. Nice catch with the from. My brain completely wiped it out. I was reading the sentence multiple times but couldn't see it
